### PR TITLE
Use developer-friendly and precise technical language in docs

### PR DESCRIPTION
**Pull request by klizhentas**
Hi Team,

While it's true that we should use unified vocabulary (similar naming of products) on the marketing website and docs, it's also true that on our developer documentation we should use crisp, precise developer language without marketing happy talk.

For example,

"Insecure secrets with true identity" is OK for enterprise website, but for engineers this creates a lot of questions.

Instead, be very specific - "Replace shared secrets with X.509 and SSH short-lived certs" is very specific, practical and avoids qualifiers like "insecure" without explaining why they are insecure.

### CODE REVIEW

File: docs/pages/index.mdx [hunk 1]
```diff
@@ -8,11 +8,11 @@ videoBanner: fSgCnQrNu78
...
 
-- Replace insecure secrets with true identity.
-- Have a single access solution for SSH, Kubernetes, databases, desktops, web applications, and more.
-- Define RBAC for every component of your infrastructure.
-- Gather audit events and session recordings to understand how users are interacting with your infrastructure.
-- Automatically onboard and offboard users with integrations such as Okta, GitHub, and Google Workspace.
+- Replace a mix of vaults, passwords, API keys and tokens with short-lived SSH and X.509 certs.
+- Have a single access point and Single-Sign-On for all your infrastructure - SSH, Kubernetes, databases, desktops, web applications, and more.
+- Write policies with Terraform or K8s CRDs for all clouds, environments and protocols and apply them with PR flow.
+- Record SSH and `kube exec` sessions, DB queries, Desktop recordings, web services and apps.
```
Comment (ibeckermayer) [inline docs/pages/index.mdx:14]:
```suggestion
- Record SSH and `kube exec` sessions, DB queries, Windows desktop sessions, and application access.
```

**Review by ibeckermayer — APPROVED** [review]
With a couple of suggestions.

File: docs/pages/index.mdx (hunk 1 above)
Comment (zmb3) [inline docs/pages/index.mdx:14]: - I would say `kubectl exec` if you're going to put it in backtick. 
- We say "record desktop recordings" which feels a bit repetitive.

File: docs/pages/index.mdx [hunk 2]
```diff
@@ -8,11 +8,11 @@ videoBanner: fSgCnQrNu78
 
 Teleport is an identity-native infrastructure access platform. With Teleport you can:
 
-- Replace insecure secrets with true identity.
-- Have a single access solution for SSH, Kubernetes, databases, desktops, web applications, and more.
-- Define RBAC for every component of your infrastructure.
-- Gather audit events and session recordings to understand how users are interacting with your infrastructure.
-- Automatically onboard and offboard users with integrations such as Okta, GitHub, and Google Workspace.
+- Replace a mix of vaults, passwords, API keys and tokens with short-lived SSH and X.509 certs.
+- Have a single access point and Single-Sign-On for all your infrastructure - SSH, Kubernetes, databases, desktops, web applications, and more.
```
Comment (zmb3) [inline docs/pages/index.mdx:12]:
```suggestion
- Have a single access point and Single-Sign-On for all of your infrastructure - SSH servers, Kubernetes clusters, databases, desktops, web applications, and more.
```

File: docs/pages/index.mdx [hunk 3]
```diff
@@ -8,11 +8,11 @@ videoBanner: fSgCnQrNu78
...
+- Write policies with Terraform or K8s CRDs for all clouds, environments and protocols and apply them with PR flow.
```
Comment (zmb3) [inline docs/pages/index.mdx:13]:
```suggestion
- Write policies with Terraform or Kubernetes resources for all clouds, environments and protocols and apply them with PR flow.
```

[the remark this answers is not in the document]
Comment (klizhentas) [reply docs/pages/index.mdx]: replaced with GitOps which is more precise term used in the industry

File: docs/pages/index.mdx (hunk 1 above)
Comment (klizhentas) [reply docs/pages/index.mdx]: applied with a couple of tweaks

Comment (klizhentas) [reply docs/pages/index.mdx]: done

**Review by zmb3 — APPROVED** [review]

**Review by evanfreed — APPROVED** [review]

**Comment by alexfornuto** [comment]
I'll manually create backport PRs since this was merged without them. I'll also apply the edits (Oxford commas) I would have applied here.

**Pull request merged**